### ISSUE TITLE
Categorical variables: queens on the board (and bishops of the color of f7)

### ISSUE THREAD

**Issue by YodaEmbedding** [gjgd/should-i-play-f6#6]
I believe the greatest influencers towards playing f3/f6 are:

 - Presence of queens
 - Presence of queens
 - No really, the presence of queens
 - Bishops of the same color as the f7 pawn

So there are three binary categorical variables to check for:

 - presence of enemy queen
 - presence of enemy f7-colored bishop
 - presence of allied f7-colored bishop

I think the idea of "never play f6" only really applied to middle game positions with these factors present. Once these factors are removed, I'll bet that rating doesn't influence whether or not f6 should be played. Can this repository demonstrate that? And establish a good solid guideline towards the playing of f6? :)

My guess for the rating discrepancy that you're currently observing is that high-rated players that play f6 usually only do once these important attackers are gone.

**Comment by gjgd** [comment]
Yes that's a good point, I agree that in general the advice "never play f6" applies the most in the middlegame, so ideally we would want to:
- remove the games where f3/f6 are played as part of an opening that is within theory
- remove the games where f3/f6 are played in an endgame.

The second point is harder because, how do you define an endgame ? Queens are certainly a good indicator, but sometimes they are traded early and you have a queen-less middlegame and sometimes they are still there in the endgame.
My idea was to remove the games where f3/f6 are played after a certain move (for example move 20), but this method also has its fault. Let me know what you think!
